docs/API.md

```diff
@@ -188,6 +188,44 @@ outerEmitter.emit('request')
 
 See the [API documentation](./interfaces/scope.html) for more details.
 
+<h2 id="decorator">Decorator</h2>
+
+A decorator is available to facilitate tracing class methods. If used on a class, all methods will be traced. Individual methods can be decorated as well. The decorator takes an optional configuration parameter that allows renaming the span and adding tags.
+
+```javascript
+const { trace } = require('dd-trace').decorator
```

Comment:
Can the existing `tracer.wrap` method be re-used for this? My thought was that it could be used for this purpose by omitting the callback, in which case it would return the decorator.

Reply:
It would also make the API more consistent with [dd-trace-py](http://pypi.datadoghq.com/trace/docs/basic_usage.html#decorator).

Reply:
From a javascript perspective, that should be very doable.
I think this type of polymorphism should play nicely with typescript as long as the callback function isn't optional in the current `tracer.wrap` implementation.

Reply:
It's not as it was actually designed with future support for decorators in mind :)

Reply:
Follow up: the callback is currently required by the TS definitions for wrap, so this should be easy to type.